src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3WebtransportManager.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Net.Quic;
+using System.Runtime.Versioning;
+using System.Net.Http.Headers;
+using System.Net.Security;
+using System.Text;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.Net.Http
+{
+    [SupportedOSPlatform("windows")]
+    [SupportedOSPlatform("linux")]
+    [SupportedOSPlatform("macos")]
+    internal sealed class Http3WebtransportManager : IDisposable, IAsyncDisposable
+    {
+        private ConcurrentDictionary<long, Http3WebtransportSession> _sessions;
+        private QuicConnection _connection;
+
+        public Http3WebtransportManager(QuicConnection connection)
+        {
+            _sessions = new ConcurrentDictionary<long, Http3WebtransportSession>();
+            _connection = connection;
+        }
+
+        public bool AddSession(QuicStream connectStream, Http3WebtransportSession webtransportSession)
+        {
+            bool ans = _sessions.TryAdd(connectStream.Id, webtransportSession);
+            return ans;
+        }
+
+        public bool FindSession(long streamId, out Http3WebtransportSession? session)
+        {
+            return _sessions.TryGetValue(streamId, out session);
+        }
+
+        public void AcceptServerStream(QuicStream stream, long sessionId)
+        {
+            Http3WebtransportSession? session;
+            _sessions.TryGetValue(sessionId, out session);
+            // if no session with that id exists throw exception
+            // https://datatracker.ietf.org/doc/html/draft-ietf-webtrans-http3#section-4
+            if (session == null)
+            {
+                stream.Abort(QuicAbortDirection.Both, (long)Http3ErrorCode.WebtransportBufferedStreamRejected);
+                return;
+            }
+            session.AcceptServerStream(stream);
+        }
+
+        public async ValueTask<QuicStream?> CreateClientStream(QuicStreamType type, long sessionId)
```

Comment:
Async methods by convention have names ending with Async, and it is better to pass a cancellation token so that the operation can be canceled from the outside